### ISSUE TITLE
Extra option to disable Green water with SADX style water Enabled?

### ISSUE THREAD

**Issue by PimpUigi** [PiKeyAr/sadx_dreamcast#229]
Hey there! I was hoping in the future someone could make an option to disable the Green Water that appears in Emerald Coast and the Station Square Sewer (and anywhere else in the game it might be) - so I could retain the SADX styled water for the oceans.

For the short term, I was hoping someone could help me write my own diff file so I can continue using SADX style water without having to suffer in these two areas of the game:

TL;DR I'd like an option for any green water to be Dreamcast water no matter what. (it'd also be fine if the sewer was the same as the ocean)
![image](https://user-images.githubusercontent.com/7344271/104551509-01cd7780-5605-11eb-8020-3197082e7cfa.png)
![image](https://user-images.githubusercontent.com/7344271/104551598-2c1f3500-5605-11eb-9450-b9ae0fe39b46.png)

**Comment by PiKeyAr** [comment]
I think the most realistic way to get close to what you want is to edit or replace the green water textures in the file `SADXWTR_WAVES.PVMX` in the mod's textures folder. There are separate textures for the sewers and the pool which you can replace with copies of other textures.
You won't be able to make it look like Dreamcast water without altering the mod's source code. At the moment there are options to enable or disable SADX style water per area. Disabling it specifically for the pool or the sewers while leaving it enabled for other water in the same area is possible in theory, but it's not something I'd consider as an option in the mod.

**Comment by PimpUigi** [comment]
_I think the most realistic way to get close to what you want is to edit or replace the green water textures in the file SADXWTR_WAVES.PVMX in the mod's textures folder. There are separate textures for the sewers and the pool which you can replace with copies of other textures._ 

That sounds good enough, and easy enough. A decent compromise. How would I go about doing that? (we could talk through it on Discord if you want, PimpUigi#3619
I respect it if you don't want to add an option for the full Dreamcast look, I mean I know it'd definitely take time and effort to edit that amount of code in for such a niche thing.

**Comment by PimpUigi** [comment]
Alright, here we go!

I think if you compare these to the Dreamcast style water, you'll find this right along the original stylistic thinking.
1. The blue tint here nearly matches the Dreamcast original, while retaining the new style of water.
![image](https://user-images.githubusercontent.com/7344271/104636991-2f9cd580-5672-11eb-99d6-315d8215f28a.png)
2. The sewer water texture matches the brighter ocean waves texture like in the Dreamcast version.
![image](https://user-images.githubusercontent.com/7344271/104637207-7d194280-5672-11eb-9d45-28ea0cf72367.png)
Good hybrid compromise, definitely prettier than the acid greens, would present a fair DX vs. SA1 argument in preference.


[SADXWTR_WAVES.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/5816492/SADXWTR_WAVES.zip)

**Comment by PimpUigi** [comment]
After playing with this for a few days, even after fixing the green water, I still prefer the Dreamcast water.
Maybe once the Bump mapping gets fixed so it can look more like the GameCube's instead of just flat. I don't know if that will ever get fixed, but at least things look good enough without the bump mapping that it's worth arguing which water is better now. : )

With what I learned here I think I can also color all the SADX water to match the hue of the Dreamcast water. If/when the bump maps get added I'll get around to doing that and we can have a cool "hybrid" option!

**Comment by PimpUigi** [comment]
I'm working on that mod to have SADX Water With Dreamcast Colors
But I've hit a snag

I don't know how to change the brighter section of the water that appears on the beach slopes:
![image](https://user-images.githubusercontent.com/7344271/104980691-2c3c7d80-59d5-11eb-9d94-32188b3637c6.png)

Perhaps if I could make them more neutral colored like the slopes of the beaches in Big's EC:
![image](https://user-images.githubusercontent.com/7344271/104982158-11b7d380-59d8-11eb-94ba-69523865747e.png)


The rest of the level is looking pretty good though
SADX Dreamcast Colors:
![image](https://user-images.githubusercontent.com/7344271/104980785-6dcd2880-59d5-11eb-8986-d840548b5ba7.png)

OG Dreamcast Colors:
![image](https://user-images.githubusercontent.com/7344271/104980737-4aa27900-59d5-11eb-80d2-a2ffe23efa64.png)

Is there a "next step" or do I just need to accept darker waters because of extra bright beaches? (this is also a problem in all iterations of Station Square, particularly the Day brand)
I fixed the Yellow evening SS water, and the extra bright SS Night water with monochrome water textures. The game pushes an extra blue layer of coding on top of them so monochrome was the way to get the game to compromise and come closest to Dreamcast coloring.

Mystic Ruins and Past water were already pretty much right, and I was able to get Crashed Egg Carrier's water the last 10% push it needed to be similar to the Dreamcast water color. Add that to the already fixed Sewer and Pond water from the start of this thread and this is fairly complete apart from this paltry Beach issue I'm having.

**Comment by PimpUigi** [comment]
I took the textures m128_sand001 through m128_sand015 and desaturated them, popped them into the SADXWTR_WAVES.pvmx and the game doesn't load them that way.
I figure I have to pop them into each of the EC and SS .pvm files, but is there an easier way to tell the game to load them instead like you've done with the SADX water?

**Comment by PiKeyAr** [comment]
SADXPC water uses additive blending for water in Emerald Coast, Adventure Fields and the Chao Gardens, which is why the seashore look brighter as multiple layers oversaturate the final color. You can only change the blending mode via code, at which point you might as well just rewrite it to use Dreamcast style models, like I do in the mod.

Anyway, this tracker is for issues and suggestions for Dreamcast Conversion, not for asking for help with mod development or posting your progress. The "SADX Style Water" option in DC Conversion is a legacy option meant for those who prefer the way the water looked in the mod before I figured out how to restore Dreamcast water. If you want to change it, it will need to be its own mod. If you need help putting together a mod, ask on x-hax.

**Comment by PimpUigi** [comment]
Oh I see. : \
Yeah I just thought I was helping you have a "closer to Dreamcast SADX Water" for _your mod._
I won't be releasing this elsewhere, at least not until/unless someone fixes bump mapping for the water. And then I guess they can use it if they're interested. When they have the option to use Dreamcast water in general though, I don't think anyone is really going to go through the effort to do all the PITA work to restore bump mapped water.

Figuring in the code adjustments needed to address the additive blending (I don't want anyone to have to go that far) I changed it to only get the water closer to Dreamcast with the additive blending in mind:
![image](https://user-images.githubusercontent.com/7344271/104988393-f2c03e00-59e5-11eb-84d4-59ab77c6b0f7.png)

You already did such amazing work, the SADX water option seemed like half a step short of completion.
Consider it a "suggestion" for your mod, obviously you're under no obligation to use it, but I thought it was helpful for those people who wanted SADX water with _Dreamcast conversion_ in mind. I found it pretty incongruent to be running around Dreamcast Converted levels and fall into a pond of Green GameCube water, or have yellow PC water pop up when we go into Twinkle Park. I understand wanting to have the prettier graphics, but those people _probably_ still want it to be accurate to the original colors. I mean I notice you had already changed the colors in general to begin with, they just felt a little all over the place. GameCube/SADX had Purple water in evening, not yellow water. The PC water in Emerald Coast was a deep blue, not as pretty as you made it, but _this_ file attached makes it closer to the Dreamcast than either of those.

Myself, I'm going to be using the Dreamcast water. Even if just to honor your hard work and diligence, which I have certainly respected the hell out of as I have re-experienced Sonic Adventure. I just figured I can't be the only one going through such experiences watching their friends play with SADX water. : \
I finished the, well, what I considered a bugfix at least. For whatever that's worth to you. 9.9
[DreamcastSADXWTR.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/5833675/DreamcastSADXWTR.zip)

**Comment by PimpUigi** [comment]
I think I'll take your advice and release it as its own mod on GameBanana or whatever.
I was definitely overzealous on my color adjustments, I see now you focused on making sure the beach sand blended in properly with the additive msand animation colors perfectly. So I reverted SS Day and Emerald Coast (which I had also fucked up with the Beach,pmvx royally it turned out - I thought that only affected Big's Emerald Coast), at least I had SS Evening+Night right, but sorry for all my nonsense.
Thanks for suggesting x-hax as well, nice people there. Sorry again.

[DreamcastSADXWTR.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/5858435/DreamcastSADXWTR.zip)
(just in case, I'd still rather it be an improvement for your Conversion)

**Comment by PaintDude** [comment]
The mod won’t work

**Comment by PaintDude** [comment]
Pls help

**Comment by PiKeyAr** [comment]
It's not going to work anymore because SADX Style Water has been removed from Dreamcast Conversion. However, it's now available as a separate mod that you can modify as you wish https://github.com/PiKeyAr/sadx-style-water

**Comment by PaintDude** [comment]
Ok thx

**Comment by PaintDude** [comment]
But I don’t know how to change textures
